build.sbt

```diff
@@ -23,8 +23,15 @@ javacOptions ++= Seq(
   "-bootclasspath", Array((java6Home / "jre" / "lib" / "rt.jar").toString, (java6Home / ".." / "Classes"/ "classes.jar").toString).mkString(File.pathSeparator)
 )
 
-// Try to future-proof scala jvm targets, in case some future scala version makes 1.7 a default
-scalacOptions += "-target:jvm-1.6"
+scalacOptions ++= (
+  if (scalaVersion.value.startsWith("2.12")) {
+    // -target is deprecated as of Scala 2.12, which uses JVM 1.8 bytecode
+    Seq.empty
+  } else {
+    // Explicitly target 1.6 for scala < 2.12
+    Seq("-target:jvm-1.6")
```

Comment:
Fwtw, with Jackson 2.7 we can require JDK 7 support if that helps (and JDK/JVM 8 obviously as needed for scala 2.12)